video_xblock/backends/html5.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+"""Html5 Video player plugin."""
+
+import json
+import re
+
+from video_xblock import BaseVideoPlayer
+
+
+class Html5Player(BaseVideoPlayer):
+    """
+    Html5Player is used for videos by providing direct URL.
+    """
+
+    url_re = re.compile(r'^(https?|ftp)://[^\s/$.?#].[^\s]*.(mpeg|mp4|ogg|webm)')
+
+    metadata_fields = []
+
+    @property
+    def advanced_fields(self):
+        """
+        Tuple of VideoXBlock fields to display in Advanced tab of edit modal window.
+
+        Hide `download_video_url` field for Html5Player.
+        """
+        return [field for field in super(Html5Player, self).advanced_fields if field != 'download_video_url']
+
+    # Html API for requesting transcripts.
+    captions_api = {}
+
+    def media_id(self, href):
+        """
+        Extract Platform's media id from the video url.
+
+        E.g. https://example.wistia.com/medias/12345abcde -> 12345abcde
+        """
+        return href
+
+    def get_type(self, href):
+        """
+        Get file extension for video.js type property.
+        """
+        return "video/" + self.url_re.search(href).groups()[-1]
+
+    def get_frag(self, **context):
+        """
+        Return a Fragment required to render video player on the client side.
+        """
+        context['data_setup'] = json.dumps({
```

Comment:
Move common `data_setup` parts shared by all backends into base class to reduce code duplication.